src/Security/Authenticator.php

```diff
@@ -117,9 +117,10 @@ public static function register_authenticator($authenticator)
             return false;
         }
 
-        if (in_array($authenticator, self::$authenticators) == false) {
-            if (call_user_func(array($authenticator, 'on_register')) === true) {
-                array_push(self::$authenticators, $authenticator);
+        $authenticators = self::config()->get('authenticators');
+        if (in_array($authenticator, $authenticators) == false) {
+            if ($authenticator::on_register() === true) {
+                self::config()->merge('authenticators', [$authenticator]);
```

Comment:
Remove the default authenticator from the Authenticators config, and then make `get_authenticators` conditionally return the list of authenticators (which is empty by default), but return a single-length default authenticator array if empty. That way as soon as you add a single item to authenticators, it disables the default.